src/HmrcTokenHandler.ts

```diff
@@ -0,0 +1,70 @@
+import { Logger } from "@aws-lambda-powertools/logger";
+import { LambdaInterface } from "@aws-lambda-powertools/commons";
+import { Constants, EnvironmentVariables } from "./utils/Constants";
+import { Metrics } from "@aws-lambda-powertools/metrics";
+import { MessageCodes } from "./models/enums/MessageCodes";
+import { AppError } from "./utils/AppError";
+import { HmrcTokenRequestProcessor } from "./services/HmrcTokenRequestProcessor";
+import { getParameter } from "./utils/Config";
+import { checkEnvironmentVariable } from "./utils/EnvironmentVariables";
+
+const { POWERTOOLS_METRICS_NAMESPACE = Constants.BAV_METRICS_NAMESPACE, POWERTOOLS_LOG_LEVEL = "DEBUG", POWERTOOLS_SERVICE_NAME = Constants.HMRC_TOKEN_LOGGER_SVC_NAME } = process.env;
+export const logger = new Logger({
+	logLevel: POWERTOOLS_LOG_LEVEL,
+	serviceName: POWERTOOLS_SERVICE_NAME,
+});
+export const metrics = new Metrics({ namespace: POWERTOOLS_METRICS_NAMESPACE, serviceName: POWERTOOLS_SERVICE_NAME });
+
+let HMRC_CLIENT_ID: string;
+
+let HMRC_CLIENT_SECRET: string;
+
+
+class HmrcTokenHandler implements LambdaInterface {
+	private readonly HMRC_CLIENT_ID_SSM_PATH = checkEnvironmentVariable(EnvironmentVariables.HMRC_CLIENT_ID_SSM_PATH, logger);
+
+	private readonly HMRC_CLIENT_SECRET_SSM_PATH = checkEnvironmentVariable(EnvironmentVariables.HMRC_CLIENT_SECRET_SSM_PATH, logger);
+
+	async handler(): Promise<void> {
+		logger.setPersistentLogAttributes({});
+		
+		try {
+			logger.info("Generating a new HMRC token");
+			if (!HMRC_CLIENT_ID) {
```

Comment:
Should these individual try catches be nested inside the parent try catch? Might end up throwing multiple errors for 1 issue cause we're catching the throw

Reply:
refactored the code to abstract the ssm fetch to a new function and letting the global try/catch to catch the AppError and action it.